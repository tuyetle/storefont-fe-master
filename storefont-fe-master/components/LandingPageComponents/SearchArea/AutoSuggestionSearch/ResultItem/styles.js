import variables from '~/styles/variables';
import themes from '~/styles/themes';

export default {
  resultItem: {
    padding: 0,
    color: variables.$colorWelcome,
  },
  resultImg: {
    margin: '0.3em',
    height: '1.25em',
    width: '2em',
  },
  resultImgHighlight: {
    ...themes['font-montserrat-light'],
    color: variables.$colorWelcome,
  },
  resultText: {
    ...themes['font-montserrat-light'],
    color: variables.$colorWelcome,
    fontSize: '0.8125em',
    marginLeft: '0.5em',
    '@media (min-width: 768px)': {
      fontSize: '0.9375em',
    },
  },
};
