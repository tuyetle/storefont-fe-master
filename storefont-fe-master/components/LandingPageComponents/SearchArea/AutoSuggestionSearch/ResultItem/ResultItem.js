import React from 'react';
import PropTypes from 'prop-types';
import match from 'autosuggest-highlight/match';
import parse from 'autosuggest-highlight/parse';
import { css } from 'glamor';
import styles from './styles';

const ResultItem = (props) => {
  const renderSuggestion = (text, query) => {
    const matches = match(text, query);
    const parts = parse(text, matches);

    return (
      parts.map((part, index) => {
        const className = part.highlight ? css(styles.resultImgHighlight) : null;

        return (
          <span className={css(className)} key={+index}>{part.text}</span>
        );
      })
    );
  };

  return (
    <div className={css(styles.resultItem)}>
      { props.imgSrc &&
      <img alt="Icon" className={css(styles.resultImg)} src={props.imgSrc} />
      }
      <span className={css(styles.resultText)}>
        {renderSuggestion(props.resultText, props.query)}
      </span>
    </div>
  );
};

ResultItem.propTypes = {
  resultText: PropTypes.string.isRequired,
  imgSrc: PropTypes.string,
  query: PropTypes.string,
};

ResultItem.defaultProps = {
  imgSrc: null,
  query: null,
};

export default ResultItem;
