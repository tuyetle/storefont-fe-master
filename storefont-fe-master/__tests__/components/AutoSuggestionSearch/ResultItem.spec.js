import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ResultItem from '~/components/LandingPageComponents/SearchArea/AutoSuggestionSearch/ResultItem/ResultItem';

describe('ResultItem', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      imgSrc: 'logo.png',
      resultText: 'title',
      query: '',
    };
    component = (<ResultItem {...props} />);
  });

  it('should have image with proper src', () => {
    const sc = shallow(component);
    expect(sc.find('img').props().src).toEqual('logo.png');
  });

  it('should match snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
