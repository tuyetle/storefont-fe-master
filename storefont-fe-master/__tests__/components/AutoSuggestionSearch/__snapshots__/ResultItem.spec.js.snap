// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem should match snapshot 1`] = `
<div
  className={
    Object {
      "data-css-lv9ezv": "",
    }
  }
>
  <img
    alt="Icon"
    className={
      Object {
        "data-css-1d68q0k": "",
      }
    }
    src="logo.png"
  />
  <span
    className={
      Object {
        "data-css-1vf9l72": "",
      }
    }
  >
    <span
      className={Object {}}
    >
      title
    </span>
  </span>
</div>
`;
